test/integration/evictions/evictions_test.go

```diff
@@ -491,7 +508,18 @@ func TestEvictionWithUnhealthyPodEvictionPolicy(t *testing.T) {
 			}
 			deleteOption := metav1.DeleteOptions{}
 			eviction := newV1Eviction(ns.Name, pod.Name, deleteOption)
-			err := clientSet.PolicyV1().Evictions(ns.Name).Evict(ctx, eviction)
+			err := clientSet.CoreV1().RESTClient().Post().
+				AbsPath("/api/v1").
+				Namespace(eviction.Namespace).
+				Resource("pods").
+				Name(eviction.Name).
+				SubResource("eviction").
+				Body(eviction).
+				// Eviction API can return http.StatusTooManyRequests (429) with retryAfterSeconds == 10s
+				// Do not retry - we want to test that the first request succeeds and make sure it doesn't unnecessarily block the test for 10s
```

Comment:
Normally the client would retry on 429 or 500+ for POST requests with `Retry-After` header:

https://github.com/kubernetes/client-go/blob/b5e4a5a9c356d800639ba18f67ae30aef96855f3/rest/with_retry.go#L309

https://github.com/kubernetes/client-go/blob/b5e4a5a9c356d800639ba18f67ae30aef96855f3/rest/request.go#L1267

which we can also send: StatusGatewayTimeout (504)

will update the description

Nevertheless we want to test the absence of the retry here.

Reply:
We need the 0 retries and not to retry on conflict to potentially catch the regression where eviction unnecessarily fails (https://github.com/kubernetes/kubernetes/issues/116552).

Also, I am not sure if it would help against connection errors, since POST requests do not normally retry on these:

https://github.com/kubernetes/client-go/blob/b5e4a5a9c356d800639ba18f67ae30aef96855f3/rest/request.go#L983

And if we observe retries generated by us:
429 - bad and we should fail
504 - possibly also bad since we use it to detect too many retries in the eviction API: https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/storage/eviction.go#L57

Reply:
My question is whether a resourceVersion conflict would be retried. If it wouldn't I don't think we should reset it.

Most clients wouldn't change the retry settings, so I don't see a point on testing something different.

Reply:
The resourceVersion conflict would be retried here:

https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/storage/eviction.go#L301

That is why we didn't catch the bug before for example in `TestEvictionWithUnhealthyPodEvictionPolicy` test where we run into this issue, so I am removing the retries there as well.

Reply:
the client would handle the retry for the `NewTooManyRequests` response

Reply:
Ahh, I understand now. I wonder why it returns such response, it doesn't sound right to me.
@liggitt ?

Reply:
the reasoning behind the 429 one: we might be trying to delete only an unready pod and if something changes to the pod (readiness or something unimportant), we give the client an option to try again and it may succeed/fail depending on the pod change

Reply:
But we only send TooManyRequests if `originalDeleteOptions.Preconditions == nil || originalDeleteOptions.Preconditions.ResourceVersion == nil`.

Reply:
yes, I think it is mostly for the client to achieve success. When eviction of the pod fails, the client might move on to the next pod and we give it a chance to evict the pod on a 2nd try.

When client sets a precondition, the request will not succeed on a 2nd try.

Reply:
Ah, I see. The bug was that we issued retries for `Preconditions.UID != nil`.

Is there an alternative way of disabling retries without having to copy the implementation of `Evict`?

Reply:
on the other hand UID conflict might not be that often, so I changed the condition to compare with the pod UID, added another if statement and comments for better readability 

PTAL @alculquicondor @liggitt

Reply:
uhm... it looks like retries are not part of the kubeconfig.

Too many changes for this PR, so expanding the implementation should be ok for now.

Reply:
I have remove the duplicated logic from evictions client in favor of this approach - looks better now